tests/pipeline/test_pipeline_algo.py

```diff
@@ -738,3 +738,62 @@ def before_trading_start(context, data):
         )
 
         self.assertTrue(count[0] > 0)
+
+
+class PipelineSequenceTestCase(WithMakeAlgo, ZiplineTestCase):
+
+    def make_algo_kwargs(self, **overrides):
+        # run the algorithm for 3 days
+        self.START_DATE = pd.Timestamp('2015-07-13', tz='utc')
+        self.END_DATE = pd.Timestamp('2015-07-15', tz='utc')
+        return self.merge_with_inherited_algo_kwargs(
+            PipelineSequenceTestCase,
+            suite_overrides=dict(
+                get_pipeline_loader=lambda column: self.fake_pipeline,
+            ),
+            method_overrides=overrides,
+        )
+
+    def fake_pipeline(self):
+        raise AssertionError
+
+    def test_pipeline_compute_before_bts(self):
+
+        # for storing and keeping track of calls to BTS and TestFactor.compute
+        trace = []
+
+        class TestFactor(CustomFactor):
+            inputs = ()
+
+            # window_length doesn't actually matter for this test case
+            window_length = 1
+
+            def compute(self, today, assets, out):
+                trace.append("CustomFactor call")
+
+        def initialize(context):
+            pipeline = attach_pipeline(Pipeline(), 'my_pipeline')
+            test_factor = TestFactor()
+            pipeline.add(test_factor, 'test_factor')
+
+        def before_trading_start(context, data):
+            pipeline_output('my_pipeline')
```

Comment:
probably worth swapping the `pipeline_output` with the `trace` so that "BTS call" will appear first in the trace if there is an error.

Reply:
Ok